### PR TITLE
IGNITE-13967 Optimize and refactor of parsing response from server.

### PR DESCRIPTION
**Pull request by ivandasch**

### CODE REVIEW

File: pyignite/stream/binary_stream.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        self.stream = BytesIO(buf) if buf else BytesIO()
+
+    @property
+    def compact_footer(self) -> bool:
+        return self.conn.client.compact_footer
+
+    @compact_footer.setter
+    def compact_footer(self, value: bool):
+        self.conn.client.compact_footer = value
+
```
Comment (isapego) [inline pyignite/stream/binary_stream.py:56]: It looks like we only need a `conn` here for binary configuration parameters. Maybe it makes sense to make a separate class for that? Like, `BinaryContext`, `BinaryMetaProvider` or something like that?

**Review by isapego — COMMENTED** [review]
Overall looks good. See my minor comments and suggestions.

File: pyignite/stream/binary_stream.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        :param conn: Connection instance, required.
+        """
+        from pyignite.connection import Connection
+
+        buf = None
+        if len(args) == 1:
+            self.conn = args[0]
+        elif len(args) == 2:
+            buf = args[0]
+            self.conn = args[1]
```
[the remark this answers is not in the document]
Comment (ivandasch) [reply pyignite/stream/binary_stream.py:39]: Yes, looks like a good idea

File: pyignite/stream/binary_stream.py (hunk 1 above)
Comment (ivandasch) [reply pyignite/stream/binary_stream.py:56]: Original author use this property as auto-discovery of compact-footer for server.
I suppose that create separate class for 1 param seems like overengineering here.

File: pyignite/connection/__init__.py [hunk 3]
```diff
@@ -97,7 +97,7 @@ def _check_ssl_params(params):
                 ).format(param))
 
     def __init__(
-        self, client: 'Client', prefetch: bytes = b'', timeout: int = None,
+        self, client: 'Client', timeout: float = 2.0,
```
[the remark this answers is not in the document]
Comment (ivandasch) [reply pyignite/connection/__init__.py:100]: Yep, this is seconds and fraction of seconds. i.e. timeout for 150 ms is 0.15

**Pull request closed without merging**